content/shmem_broadcast.tex

```diff
@@ -72,12 +72,15 @@
     needed to ensure this): The \VAR{pSync} array on all \ac{PE}s in the
     \activeset{} is not still in use from a prior call to a broadcast routine.  The
     \dest{} array on all \ac{PE}s in the \activeset{} is ready to accept the
-    broadcast data.
+    broadcast data. \newtext{The data in the \source{} array on the root \ac{PE} is ready to be broadcasted.}
```

Comment:
This unit test illustrates the problem: https://github.com/openshmem-org/tests-cray/blob/master/sma1/shmem_broadcast_all.c

It's often easy for applications to resolve the data dependence involved with the destination buffer.  However, this change would introduce a race into the above test since there is no synchronization between the producer PE setting up the source buffer the consumer PEs calling the barrier.

A further point - this doesn't restrict implementations from using get operations to implement broadcast.  It restricts them from using _unsynchronized_ gets.  Unsynchronized gets would be most useful in an implementation that uses a full fan-out from the root PE, which is not a scalable algorithm.  In any hierarchical broadcast based on gets, you would still need to synchronize the RAW dependence between the interior nodes in the tree regardless of this semantic.  Thus the performance benefit is limited to nodes that are children of the root, which will vanish at large enough scale.